frontend/public/components/secrets/subforms.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+
+const generateSecret = () => {
+  // http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
+  const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+  return s4() + s4() + s4() + s4();
+};
+
+export class WebHookSecretSubform extends React.Component<WebHookSecretSubformProps, WebHookSecretSubformState> {
+  constructor(props) {
+    super(props);
+    this.state = {WebHookSecretKey: this.props.stringData.WebHookSecretKey || ''};
```

Comment:
nit: I'd call this `webhookSecretKey`. State properties usually begin with lowercase letters